google-http-client/src/main/java/com/google/api/client/http/UrlEncodedContent.java

```diff
@@ -42,18 +42,33 @@
  *
  * <p>Implementation is not thread-safe.
  *
- * @since 1.0
  * @author Yaniv Inbar
+ * @since 1.0
  */
 public class UrlEncodedContent extends AbstractHttpContent {
 
   /** Key name/value data. */
   private Object data;
 
+  /** Use URI Path encoder flag. False by default (use legacy and deprecated escapeUri) */
+  private Boolean useEscapeUriPathEncoding;
+
   /** @param data key name/value data */
   public UrlEncodedContent(Object data) {
     super(UrlEncodedParser.MEDIA_TYPE);
     setData(data);
+    this.useEscapeUriPathEncoding = false;
+  }
+
+  /**
+   * @param data key name/value data
+   * @param useEscapeUriPathEncoding replace the legacy (and deprecated) URI encoder, by the
```

Comment:
describe this in terms of behavior rather than implementation. The user doesn't care which encoder is used, only how URIs are encoded.

Reply:
I tried to improve it. Any feedback is welcomed!!